src/Microsoft.AspNetCore.Mvc.TagHelpers/PartialTagHelper.cs

```diff
@@ -88,7 +108,7 @@ public class PartialTagHelper : TagHelper
         // Internal for testing
         internal object ResolveModel()
         {
-            if (Model != null & For != null)
+            if (_hasFor && _hasModel)
```

Comment:
Add comments about _not_ falling back to `ViewContext.ViewData.Model` when `Model` is explicitly set to `null`.

Also correct the comment above that fallback to say "were not set" instead of "are null".